### ISSUE TITLE
TaskManager: can we collapse non-active projects?

### ISSUE THREAD

**Issue by Thanood** [monterey-framework/monterey#287]
Can we collapse non-active projects? In other words, can TaskManager open like this:

![image](https://cloud.githubusercontent.com/assets/677826/18379764/723f0a08-7674-11e6-951f-cf695bd2b194.png)

I think it's much more readable when only the tasks for the current project are opened.
Also, It would be nice if TaskManager remembered what I've collapsed/opened when I close and reopen it.

**Comment by JeroenVinke** [comment]
Definitely something we need to improve. Nikolaj has mentioned that he has a couple of ideas, so lets see what he comes up with. Fyi: I added the "Start new task" node because I wasn't sure if it's clear enough that you can start new tasks by clicking on the project name.

**Comment by JeroenVinke** [comment]
Let's use this issue to discuss improvements to the taskmanager

From https://github.com/monterey-framework/monterey/issues/264

> I feel sometimes that this grid view is too crowded, so highlighting the running task(s) is my best navigation help here.

> How about setting show finished tasks false by default (as I am still not sure why would I need to see the more than a proof of a successful execution of the end task command. Once I see that, since i cannot restart them (not to say that can find the need for such activation) I would like to remove them completely.

> As the projects are added according to add below last rule, I am not sure that any sorting action done by Monterey would be of value.

> If the list of projects is at the order of 10+ - we would need to offer alphabetical and last accessed time as criteria - just as I always to with Files Explorer. To mimic the "real" Task manager we should also consider sorting by running time and eventually by resources usage when we get the access to basic OS API.

**Comment by JeroenVinke** [comment]
![image](https://cloud.githubusercontent.com/assets/2189477/18607420/1a5b7368-7ccc-11e6-8af0-bfd3bbd487f1.png)

1: Workflow editor does not need to be in taskmanager. We can have a tile that leads to a workflow editor screen. Brings down complexity of the right side of the taskmanager
2: Can we add favorite commands to the tiles outside of the taskmanager? Why do we need a second collection of tiles? We could create a Command category (https://github.com/monterey-framework/monterey/issues/297)
3: Often useless info, resulting in crowded feeling
4: I probably don't need to see all these projects. Can we somehow have "active" and "inactive" projects?

**Comment by Thanood** [comment]
1. I agree completely.
2. Yes, but maybe have them at the top of the task list only and let the users decide which tasks they want as tiles. Favorite them with a small star icon button (outline/filled showing fav status). I think that's easier to use (one click less).
3. I agree. Maybe start the task manager with "Show finished" unchecked and inactive projects collapsed.
4. IMO an "active" project is the currently selected one and "inactive" projects are the others. 😃 We could add a "hide inactive" projects. **OR** toggle them active/inactive from the project list on main screen.
5. I think task manager should save project states (collapsed/show finished) when the user closes it.

**Comment by JeroenVinke** [comment]
![image](https://cloud.githubusercontent.com/assets/2189477/18608854/e097883c-7cf4-11e6-98be-80df42562009.png)

![image](https://cloud.githubusercontent.com/assets/2189477/18608857/e3409484-7cf4-11e6-8f4b-bb24a8070d6d.png)

![image](https://cloud.githubusercontent.com/assets/2189477/18608858/e605d9ae-7cf4-11e6-86cf-0d8cce634081.png)

**Comment by JeroenVinke** [comment]
The above was made in paint by the way

**Comment by adriatic** [comment]
Coming late to the party ... 

- I like all critical comments Jeroen wrote [above](https://github.com/monterey-framework/monterey/issues/287#issuecomment-247760332).
- Do not understand some of Daniel's [responses](https://github.com/monterey-framework/monterey/issues/287#issuecomment-247763788), so cannot state my opinion on these
- Definitively like Jeroen's [solution to reduce crowding](https://github.com/monterey-framework/monterey/issues/287#issuecomment-247776427) - as **this was my proposal few days back**, where I mentioned classification into two categories: `tools` and `utilities`. I worry about using "unusual" names like `launchers` because I discovered a few days ago that I had wrong understanding of "who is who". While we are certainly making big steps in improving Monterey's utility for "experienced users", we still need to find the way to reduce the "information overhead" for beginners.

How about introducing the concept of "gears", having first gear as the default which exposes the minimum to get going and leaving it to the user to switch to higher gears as she gains the confidence in driving Monterey?

**Comment by JeroenVinke** [comment]
>  I worry about using "unusual" names like launchers because I discovered a few days ago that I had wrong understanding of "who is who".

I do not think that using "Launchers" as category name will make it worse, I think it will make it better. If you are in the "Launchers" category, and you only see things like "CMD", "Explorer"  and "Chrome" then you get a sense of what a launcher is. I bet that one of the first things you'll do as a user is to click on all the categories to see what monterey has to offer

**Comment by adriatic** [comment]
Cannot disagree with your view either - particularly since all of these tiles will be "instrumented" with tooltip based help

**Comment by vegarringdal** [comment]
>3: Often useless info, resulting in crowded feeling

"Show finished" unchecked, looks very crowded

>4: I probably don't need to see all these projects. Can we somehow have "active" and "inactive" projects?

What about a checkbox on top for "show selected project or just show active" and it shows just the current selected project in the project list

**Comment by JeroenVinke** [comment]
Yes, we could hide finished tasks by default (nr 3 in @Thanood's comment). 

And as @vegarringdal said, we can have two radio buttons: show all projects and show selected project, where "show selected project" is checked by default. When "show selected project" is checked, it will only show the project you selected in the project list:

![image](https://cloud.githubusercontent.com/assets/2189477/18609515/34ebb55c-7d04-11e6-9e4d-2374486d5608.png)

**Comment by adriatic** [comment]
I like this "we can have two radio buttons: show all projects and show selected project" as the proposed default leads to my preference "not to crowd the task manager with items that should appear more globally". When I proposed to create Monterey Task manager, my mental model was the OS Task manager - just a lot simpler, as Monterey is not as complex as a typical OS :-)

The interplay between the `project list` collection (defined at the top level) and the Task manager's `list of items` is not optimal, mostly because of the lack of clear definition of the Task manager's items "type". This definition would be easier to set if we know that the Task manager **always** shows just the running applications like

![image](https://cloud.githubusercontent.com/assets/2712405/18609657/18afd5be-7cd5-11e6-989d-a72336e08e18.png)

using the tabbed interface (which is seemingly being added - at least by using Paint 😏 ) to allow some other views of the running apps.

I believe that this is correctly recognized as the key remaining item in the current Monterey version and it is no better time to address this than now, when we are all thinking about it.

As I live in a different physical reality pupulated with people that cannot think as fast as Lieutenant Data (Startrek Second Generation) if I am expected to help with this - I need more time to think

**Comment by JeroenVinke** [comment]
I have moved the workflow editor outside of the taskmanager. I have removed the "Start new task" node, as you can click on the name of the project instead. Finished tasks are now hidden by default, but you may have to disable this setting from the preferences, as it only affects new sessions
